libpimeval/src/pimParamsPerf.h

```diff
@@ -23,25 +23,42 @@ class pimParamsPerf
   pimParamsPerf(pimParamsDram* paramsDram);
   ~pimParamsPerf() {}
 
-  double getMsRuntimeForBytesTransfer(uint64_t numBytes) const;
-  double getMsRuntimeForFunc1(PimCmdEnum cmdType, const pimObjInfo& obj) const;
-  double getMsRuntimeForFunc2(PimCmdEnum cmdType, const pimObjInfo& obj) const;
-  double getMsRuntimeForRedSum(PimCmdEnum cmdType, const pimObjInfo& obj, unsigned numPass) const;
-  double getMsRuntimeForBroadcast(PimCmdEnum cmdType, const pimObjInfo& obj) const;
-  double getMsRuntimeForRotate(PimCmdEnum cmdType, const pimObjInfo& obj) const;
+  class perfEnergy
+  {
+    public:
+      perfEnergy() : m_msRuntime(0.0), m_mjEnergy(0.0) {}
+      perfEnergy(double msRuntime, double mjEnergy) : m_msRuntime(msRuntime), m_mjEnergy(mjEnergy) {}
+
+      double m_msRuntime;
+      double m_mjEnergy;
+  };
+
+  perfEnergy getPerfEnergyForBytesTransfer(uint64_t numBytes) const;
+  perfEnergy getPerfEnergyForFunc1(PimCmdEnum cmdType, const pimObjInfo& obj) const;
+  perfEnergy getPerfEnergyForFunc2(PimCmdEnum cmdType, const pimObjInfo& obj) const;
+  perfEnergy getPerfEnergyForRedSum(PimCmdEnum cmdType, const pimObjInfo& obj, unsigned numPass) const;
+  perfEnergy getPerfEnergyForBroadcast(PimCmdEnum cmdType, const pimObjInfo& obj) const;
+  perfEnergy getPerfEnergyForRotate(PimCmdEnum cmdType, const pimObjInfo& obj) const;
 
 private:
-  double getMsRuntimeBitSerial(PimDeviceEnum deviceType, PimCmdEnum cmdType, PimDataType dataType, unsigned bitsPerElement, unsigned numPass) const;
+  perfEnergy getPerfEnergyBitSerial(PimDeviceEnum deviceType, PimCmdEnum cmdType, PimDataType dataType, unsigned bitsPerElement, unsigned numPass) const;
 
   const pimParamsDram* m_paramsDram;
   double m_tR; // Row read latency in ms
   double m_tW; // Row write latency in ms
   double m_tL; // Logic operation / tCCD in ms
   double m_fulcrumAluLatency = 0.00000609; // 6.09ns
+  
   unsigned m_flucrumAluBitWidth = 32;
   double m_blimpCoreLatency = 0.000005; // 200 MHz. Reference: BLIMP paper
   unsigned m_blimpCoreBitWidth = 64; 
 
+  double m_eR; // Row read(ACT) energy in mJ microjoule
+  double m_eL; // Logic energy in mJ microjoule
+  double m_pB; // background power in W
+  double m_fulcrumAluEnergy = m_fulcrumAluLatency * 999999999.9; // todo: ALU power
```

Comment:
@fzx333 this is different for different operations.
For Arithmetic operations, the ALU energy consumption is `0.0004992329586 nJ`
For logical operations this is `0.0001467846411 nJ`
```suggestion
  double m_fulcrumALUArithmeticEnergy = 0.0004992329586; // in nJ taken from fulcrum paper
   double m_fulcrumALULogicalEnergy = 0.0001467846411; //in nJ
```

Reply:
Here mJ = mili joules